accepted/future-releases/nnbd/feature-specification.md

```diff
@@ -901,6 +904,56 @@ These are extended as
 per
 [separate proposal](https://github.com/dart-lang/language/blob/master/resources/type-system/flow-analysis.md).
 
+### The main function
+
+The section 'Scripts' in the language specification is replaced by the
+following:
+
+Let _L_ be a library that exports a declaration _D_ named `main`.  It is a
```

Comment:
This includes setters, right?

Reply:
No, that would not apply to setters, because the spec insists that their name ends in `=`. It may seem silly that we can have a setter like `set main(_) {}` when we can't have `Function get main => ...;` but it is a consequence of the existing terminology, and I don't think it causes any harm.

Reply:
Since no user thinks in terms of this, wouldn't it be more coherent just to ban 'main=' as well? (Dart tells me that 'main' must be a function declaration but I'm still allowed to create a setter).

Reply:
I added this proposal in a comment on #1248. It's in principle a breaking change, but I agree that it is confusing that `set main(_) {}` is allowed.

Reply:
It's only breaking in null safe code, right? We'll still allow whatever in legacy.

Reply:
True, this is a null-safety only update.